tests/framework/extensions/clusters/clusters.go

```diff
@@ -340,3 +436,37 @@ func CreateK3SRKE2Cluster(client *rancher.Client, rke2Cluster *apisV1.Cluster) (
 
 	return cluster, nil
 }
+
+func UpdateK3SRKE2Cluster(client *rancher.Client, cluster *v1.SteveAPIObject, updatedCluster *apisV1.Cluster) (*v1.SteveAPIObject, error) {
+	updateCluster, err := client.Steve.SteveType(ProvisioningSteveResouceType).ByID(cluster.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	updatedCluster.ObjectMeta.ResourceVersion = updateCluster.ObjectMeta.ResourceVersion
+
+	cluster, err = client.Steve.SteveType(ProvisioningSteveResouceType).Update(cluster, updatedCluster)
+	if err != nil {
+		return nil, err
+	}
+
+	err = kwait.Poll(500*time.Millisecond, 2*time.Minute, func() (done bool, err error) {
+		client, err = client.ReLogin()
+		if err != nil {
+			return false, err
+		}
+
+		_, err = client.Steve.SteveType(ProvisioningSteveResouceType).ByID(cluster.ID)
```

Comment:
Would it be wise to make sure the cluster is in active state as well.
Doing something like 
```
clusterResp err = client.Steve.SteveType(ProvisioningSteveResouceType).ByID(cluster.ID)
clusterResp.ObjectMeta.State.Name == "Active"

```

just checking the condition above.

Reply:
Yeah, that's a good idea. I'll make that change so we can make sure it's in a state of `Active` after the changes are applied.